File: mobile/lib/domain/entities/lesson.dart
import 'question.dart';

enum LessonStepType {
  concept,
  formula,
  visualExample,
  audioSimulation,
  interactiveQuestion,
}

class LessonFormulaData {
  final String latex;
  final String description;
  final List<Map<String, String>> variables;

  const LessonFormulaData({
    required this.latex,
    required this.description,
    required this.variables,
  });
}

class LessonVisualModelData {
  final String type; // 'function_graph' | 'audio_wave'
  final String? functionExpr;
  final List<Map<String, dynamic>>? points;
  final String? audioTranscript;
  final String? speakerName;

  const LessonVisualModelData({
    required this.type,
    this.functionExpr,
    this.points,
    this.audioTranscript,
    this.speakerName,
  });
}

class LessonStep {
  final String id;
  final int stepNumber;
  final LessonStepType type;
  final String title;
  final String? subtitle;
  final String content;
  final List<String>? highlightNotes;
  final LessonFormulaData? formulaData;
  final LessonVisualModelData? visualModelData;
  final Question? interactiveQuestion;

  const LessonStep({
    required this.id,
    required this.stepNumber,
    required this.type,
    required this.title,
    this.subtitle,
    required this.content,
    this.highlightNotes,
    this.formulaData,
    this.visualModelData,
    this.interactiveQuestion,
  });
}

class Lesson {
  final String id;
  final String courseId;
  final String skillId;
  final String title;
  final int estimatedMinutes;
  final String summary;
  final List<LessonStep> steps;
  final Question reinforcementExercise;

  const Lesson({
    required this.id,
    required this.courseId,
    required this.skillId,
    required this.title,
    required this.estimatedMinutes,
    required this.summary,
    required this.steps,
    required this.reinforcementExercise,
  });
}

File: src/domain/entities/Lesson.ts
import { Question } from './Question';

export type LessonStepType = 'concept' | 'formula' | 'visual_example' | 'audio_simulation' | 'interactive_question';

export interface LessonStep {
  id: string;
  stepNumber: number;
  type: LessonStepType;
  title: string;
  subtitle?: string;
  content: string;
  highlightNotes?: string[];
  formulaData?: {
    latex: string;
    description: string;
    variables: { symbol: string; meaning: string }[];
  };
  visualModelData?: {
    type: 'function_graph' | 'audio_wave';
    functionExpr?: string; // e.g. "2x + 3"
    points?: { x: number; y: number; label?: string }[];
    audioTranscript?: string;
    speakerName?: string;
  };
  interactiveQuestion?: Question;
}

export interface Lesson {
  id: string;
  courseId: string;
  skillId: string;
  title: string;
  estimatedMinutes: number;
  summary: string;
  steps: LessonStep[];
  reinforcementExercise: Question;
}
